Classes/Outlet/OutletArgument.php

```diff
@@ -0,0 +1,263 @@
+ */
+
+
+use TYPO3\CMS\Extbase\Mvc\Controller\MvcPropertyMappingConfiguration;
+use TYPO3\CMS\Extbase\Property\Exception\TargetNotFoundException;
+use TYPO3\CMS\Extbase\Utility\TypeHandlingUtility;
+use TYPO3\CMS\Extbase\Validation\Exception\NoSuchValidatorException;
+use TYPO3\CMS\Extbase\Validation\Validator\ConjunctionValidator;
+use TYPO3\CMS\Extbase\Validation\ValidatorResolver;
+
+/**
+ * A outlet argument
+ *
+ * @api
+ */
+class OutletArgument
+{
+    /**
+     * @var \TYPO3\CMS\Extbase\Property\PropertyMapper
+     */
+    protected $propertyMapper;
+
+    /**
+     * @var MvcPropertyMappingConfiguration
+     */
+    protected $propertyMappingConfiguration;
+
+    /**
+     * @var \TYPO3\CMS\Extbase\Validation\ValidatorResolver
+     * @inject
+     */
+    protected $validatorResolver;
+
+    /**
+     * Name of this argument
+     *
+     * @var string
+     */
+    protected $name = '';
+
+    /**
+     * Data type of this argument's value
+     *
+     * @var string
+     */
+    protected $dataType = null;
+
+    /**
+     * Actual value of this argument
+     *
+     * @var mixed
+     */
+    protected $value = null;
+
+    /**
+     * A custom validator, used supplementary to the base validation
+     *
+     * @var array<\TYPO3\CMS\Extbase\Validation\Validator\ValidatorInterface>
+     */
+    protected $validators = null;
+
+    /**
+     * The validation results. This can be asked if the argument has errors.
+     *
+     * @var \TYPO3\CMS\Extbase\Error\Result
+     */
+    protected $validationResults = null;
+
+    /**
+     * @param \TYPO3\CMS\Extbase\Property\PropertyMapper $propertyMapper
+     */
+    public function injectPropertyMapper(\TYPO3\CMS\Extbase\Property\PropertyMapper $propertyMapper)
+    {
+        $this->propertyMapper = $propertyMapper;
+    }
+
+    /**
+     * @param \TYPO3\CMS\Extbase\Mvc\Controller\MvcPropertyMappingConfiguration $propertyMappingConfiguration
+     */
+    public function injectPropertyMappingConfiguration(MvcPropertyMappingConfiguration $propertyMappingConfiguration)
+    {
+        $this->propertyMappingConfiguration = $propertyMappingConfiguration;
+    }
+
+    /**
+     * @param \TYPO3\CMS\Extbase\Validation\ValidatorResolver $validatorResolver
+     */
+    public function injectValidatorResolver(ValidatorResolver $validatorResolver)
+    {
+        $this->validatorResolver = $validatorResolver;
+    }
+
+    /**
+     * Constructs this controller argument
+     *
+     * @param string $name Name of this argument
+     * @param string $dataType The data type of this argument
+     * @throws \InvalidArgumentException if $name is not a string or empty
+     * @api
+     */
+    public function __construct($name, $dataType)
+    {
+        if (!is_string($name)) {
+            throw new \InvalidArgumentException('$name must be of type string, ' . gettype($name) . ' given.', 1187951688);
+        }
+        if ($name === '') {
```

Comment:
This should be `empty($name)` since we don't allow `'0'` as argument name (right?)